sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/ExecutorServiceParallelExecutor.java

```diff
@@ -304,8 +304,9 @@ public void run() {
             visibleUpdates.offer(VisibleExecutorUpdate.fromThrowable(update.getException().get()));
           }
         }
-        fireTimers();
-        mightNeedMoreWork();
+        if (!fireTimers()) {
```

Comment:
Add a comment here, since it's not immediately clear what "fireTimers" returns. Eg:
```
if (!fireTimers()) {
  // If any timers fired, then we may have more work.
  ...
}
```

(Or is it if there are unfired timers? etc.)